tests/jballerina-integration-test/src/test/resources/configurables/configStructuredTypesProject/configStructuredTypes/modules/mod1/mod1.bal

```diff
@@ -47,3 +47,20 @@ public type County  record {
 
 public type PersonArray Person;
 
+public type Lecturer record {|
+    string name;
+    Department department;
+|};
+
+public type Department readonly & record {|
+    string name;
+|};
+
+public type Lawyer readonly & record {|
```

Comment:
shall we have test with optional and read only fileds too. https://github.com/ballerina-platform/ballerina-lang/issues/30200